flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/AdaptiveSchedulerTest.java

```diff
@@ -1080,10 +1053,10 @@ private Iterable<RootExceptionHistoryEntry> runExceptionHistoryTests(
                             declarativeSlotPool,
                             createSlotOffersForResourceRequirements(
                                     ResourceCounter.withResource(
-                                            ResourceProfile.UNKNOWN, numAvailableSlots)),
+                                            ResourceProfile.UNKNOWN, PARALLELISM)),
                             taskManagerGateway);
                 });
-        listener.waitForRunning();
+        taskManagerGateway.waitForSubmissions(4, TestingUtils.infiniteDuration());
```

Comment:
```suggestion
        taskManagerGateway.waitForSubmissions(PARALLELISM, TestingUtils.infiniteDuration());
```
We can remove the magic number here as well...